sys/net/gnrc/routing/rpl/gnrc_rpl.c

```diff
@@ -423,6 +423,11 @@ uint8_t gnrc_rpl_gen_instance_id(bool local)
 
 void gnrc_rpl_configure_root(gnrc_netif_t *netif, const ipv6_addr_t *dodag_id)
 {
+    /* only run RPL on wireless interfaces */
+    if (gnrc_netapi_get(netif->pid, NETOPT_IS_WIRED, 0, NULL, 0) == 1) {
+        return;
+    }
```

Comment:
Mh... I know defacto it doesn't make sense to run RPL over a wired network, but e.g. for virtualization this might still be important (e.g. many testings of RPL were done with DESVirt on `native`' TAP if I remember correctly). Does it maybe make more sense to make this dependent on other factors e.g. if a the given OFs can be calculated?

Reply:
>  many testings of RPL were done with DESVirt on `native` TAP if I remember correctly

Doesn't it make much more sense to use ZEP for that?
Actually this whole PR only exists to enable testing of RPL on CI using ZEP - #17353

Reply:
> Actually this whole PR only exists to enable testing of RPL on CI using ZEP - #17353

I am not really clear, how this would help.

Reply:
The idea is that adding a prefix via `nib prefix add` should behave the same as adding a prefix via router advertisement or adding a prefix via UHCP / DHCPv6.

Those will also init RPL on those interfaces, but when adding this I thought it would make sense no to do so for wired interfaces.

I can drop that commit if you prefer that.

Reply:
> The idea is that adding a prefix via `nib prefix add` should behave the same as adding a prefix via router advertisement or adding a prefix via UHCP / DHCPv6.
> 
> Those will also init RPL on those interfaces, but when adding this I thought it would make sense no to do so for wired interfaces.

Mh maybe then rather `nib prefix add` and `rpl root` should remain separate commands and not everything should be automated ;-).

Reply:
There is now #17709

Reply:
I have no experience with RPL, but quickly searching for `wireless` in RFC65550 shows to me that it is not strictly restricted to wireless interfaces.

> [1.2](https://datatracker.ietf.org/doc/html/rfc6550#section-1.2).  Expectations of Link-Layer Type

> In compliance with the layered architecture of IP, RPL does not rely
> on any particular features of a specific link-layer technology.  RPL
> is designed to be able to operate over a variety of different link
> layers, including ones that are constrained, potentially lossy, or
> typically utilized in conjunction with highly constrained host or
> router devices, such as but not limited to, low-power wireless or PLC
> (Power Line Communication) technologies.